security/sec-iride/src/main/java/org/geoserver/security/iride/IrideUserGroupService.java

```diff
@@ -0,0 +1,183 @@
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+package org.geoserver.security.iride;
+
+import java.io.IOException;
+import java.util.SortedSet;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.geoserver.security.GeoServerUserGroupService;
+import org.geoserver.security.GeoServerUserGroupStore;
+import org.geoserver.security.config.SecurityNamedServiceConfig;
+import org.geoserver.security.impl.AbstractUserGroupService;
+import org.geoserver.security.impl.GeoServerUser;
+import org.geoserver.security.impl.GeoServerUserGroup;
+import org.geotools.util.logging.Logging;
+
+/**
+ * <code>GeoServer</code> user group security service, backed by <code>CSI</code> <code>IRIDE</code> service.
+ *
+ * @author "Simone Cornacchia - seancrow76@gmail.com, simone.cornacchia@consulenti.csi.it (CSI:71740)"
+ */
+public class IrideUserGroupService extends AbstractUserGroupService implements GeoServerUserGroupService {
+
+    /**
+     * Logger.
+     */
+    private static final Logger LOGGER = Logging.getLogger(IrideUserGroupService.class);
+
+    /**
+     * Constructor.
+     */
+    public IrideUserGroupService() {
+        super();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.geoserver.security.impl.AbstractGeoServerSecurityService#initializeFromConfig(org.geoserver.security.config.SecurityNamedServiceConfig)
+     */
+    @Override
+    public void initializeFromConfig(SecurityNamedServiceConfig config) throws IOException {
+        LOGGER.log(Level.INFO,
+            "Initializing {0} with configuration object: \n\t {1}",
+            new Object[] { this.getClass().getSimpleName(), config }
+        );
+
+        this.name = config.getName();
+
+        // TODO: check if needed and, if so, implement
+        super.initializeFromConfig(config);
+    }
+
+    /**
+     * <code>IRIDE</code> user group service is <em>read-only</em>, therefore there is no support for {@link GeoServerUserGroupStore}.
+     */
+    /*
+     * (non-Javadoc)
+     * @see org.geoserver.security.impl.AbstractGeoServerSecurityService#canCreateStore()
+     */
+    @Override
+    public boolean canCreateStore() {
+        return false;
+    }
+
+    /**
+     * <code>IRIDE</code> user group service is <em>read-only</em>, therefore there is no support for {@link GeoServerUserGroupStore}.
+     */
+    /*
+     * (non-Javadoc)
+     * @see org.geoserver.security.impl.AbstractUserGroupService#createStore()
+     */
+    @Override
+    public GeoServerUserGroupStore createStore() throws IOException {
+        return null;
```

Comment:
As far as I understand, returning `null` is the way the UserGroupService signals to its client that it doesn't support the creation of a store.

The reason I've preferred to explicitly override those two methods is to stress the fact that no store is supported.

Even if not expressly needed, for the aforementioned reason I'd rather keep those two overrided methods.